internal/testing/cluster.go

```diff
@@ -29,17 +36,35 @@ type TestCluster struct {
 	// Autopilot *autopilot.Client // TODO: add once available
 
 	cleanups  []func() error
-	listeners []net.Listener
 	shutdowns []func(context.Context) error
 
-	dir string
-	wg  sync.WaitGroup
+	miner       *node.Miner
+	dir         string
+	gatewayAddr string
+	wg          sync.WaitGroup
 }
 
+// randomPassword creates a random 32 byte password encoded as a string.
 func randomPassword() string {
 	return hex.EncodeToString(frand.Bytes(32))
 }
 
+// Retry will call 'fn' 'tries' times, waiting 'durationBetweenAttempts'
+// between each attempt, returning 'nil' the first time that 'fn' returns nil.
+// If 'nil' is never returned, then the final error returned by 'fn' is
+// returned.
+func Retry(tries int, durationBetweenAttempts time.Duration, fn func() error) (err error) {
```

Comment:
the existence of this function makes me sad. I would prefer to discourage its use as much as possible.